src/ConnectedMode.UnitTests/ServerEventFilteredChannelTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents.Issues;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents.TaintVulnerabilities; 
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests
+{
+    // since TaintServerEventChannel and IssueChangedServerEventChannel only differ in the value of the type parameter and in mef export contract, the tests results are the same for both
```

Comment:
I tried making the test class generic, so it could run the tests for both types, but vstest does not support that (unlike NUnit). I also couldn't find a way to get rid of the 2 implementation classes and merge them into one because of MEF

Reply:
👍 
You can delete the comment though, as from the perspective of this class, it shouldn't know that taint and issue event exist.

Reply:
@georgii-borovinskikh-sonarsource delete the comment?